go/plugins/vertexai/vertexai.go

```diff
@@ -180,35 +191,72 @@ func NewGenerator(ctx context.Context, model, projectID, location string) (ai.Ge
 }
 
 // convertParts converts a slice of *ai.Part to a slice of genai.Part.
-func convertParts(parts []*ai.Part) []genai.Part {
+func convertParts(parts []*ai.Part) ([]genai.Part, error) {
 	res := make([]genai.Part, 0, len(parts))
 	for _, p := range parts {
-		res = append(res, convertPart(p))
+		part, err := convertPart(p)
+		if err != nil {
+			return nil, err
+		}
+		res = append(res, part)
 	}
-	return res
+	return res, nil
 }
 
 // convertPart converts a *ai.Part to a genai.Part.
-func convertPart(p *ai.Part) genai.Part {
+func convertPart(p *ai.Part) (genai.Part, error) {
 	switch {
 	case p.IsText():
-		return genai.Text(p.Text)
+		return genai.Text(p.Text), nil
 	case p.IsMedia():
-		return genai.Blob{MIMEType: p.ContentType, Data: []byte(p.Text)}
+		url := p.Text
```

Comment:
Can this part be deduped with the googleai code? Perhaps it returns (string, []byte).
It could live in plugins/internal.
Also I think it's complicated enough to warrant a test.

Reply:
Done.